megatron/model/utils.py

```diff
@@ -402,3 +402,16 @@ def mark_norms_for_sequence_parallel_grad_sync(module, neox_args):
             for name, param in module_.named_parameters():
                 if param.requires_grad:
                     param.register_hook(reduce_weight_grads_from_model_parallel_region)
+
+
+def linear_implementation_router(neox_args):
```

Comment:
I suggest removing `router` so that it can't be confused with MoE routers.

Reply:
Updated to `get_parallel_linear`